base/linalg/symmetric.jl

```diff
@@ -107,6 +107,21 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    i == j || throw(ArgumentError("Cannot set a non-diagonal index in a symmetric matrix"))
```

Comment:
Only being able to assign to the upper or lower triangle depending on the value of `uplo` also breaks the abstraction; the user shouldn't have to care about which triangle is stored since all they want to know about it is that it's symmetric.